src/common/libjobspec/jobspec.cpp

```diff
@@ -0,0 +1,327 @@
+ *  under the terms of the GNU General Public License as published by the Free
+ *  Software Foundation; either version 2 of the license, or (at your option)
+ *  any later version.
+ *
+ *  Flux is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the IMPLIED WARRANTY OF MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the terms and conditions of the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ *  See also:  http://www.gnu.org/licenses/
+\*****************************************************************************/
+
+#include "jobspec.hpp"
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+using namespace std;
+using namespace Flux::Jobspec;
+
+void Resource::parse_yaml_count (const YAML::Node &cnode)
+{
+    /* count can have an unsigned interger value */
+    if (cnode.IsScalar()) {
+        count.min = cnode.as<unsigned>();
+        count.max = count.min;
+        return;
+    }
+
+    /* or count may be a more complicated verbose form */
+    if (!cnode.IsMap()) {
+        throw invalid_argument ("count is not a mapping");
+    }
+
+    /* Verify existance of required entries */
+    if (!cnode["min"]) {
+        throw invalid_argument ("Key \"min\" missing from count");
+    }
+    if (!cnode["min"].IsScalar()) {
+        throw invalid_argument ("Value of \"min\" must be a scalar");
+    }
+    if (!cnode["max"]) {
+        throw invalid_argument ("Key \"max\" missing from count");
+    }
+    if (!cnode["max"].IsScalar()) {
+        throw invalid_argument ("Value of \"max\" must be a scalar");
+    }
+    if (!cnode["operator"]) {
+        throw invalid_argument("Key \"operator\" missing from count");
+    }
+    if (!cnode["operator"].IsScalar()) {
+        throw invalid_argument ("Value of \"operator\" must be a scalar");
+    }
+    if (!cnode["operand"]) {
+        throw invalid_argument ("Key \"operand\" missing from count");
+    }
+    if (!cnode["operand"].IsScalar()) {
+        throw invalid_argument ("Value of \"operand\" must be a scalar");
+    }
+
+    /* Validate values of entries */
+    count.min = cnode["min"].as<unsigned>();
+    if (count.min < 1) {
+        throw invalid_argument ("\"min\" must be greater than zero");
+    }
+
+    count.max = cnode["max"].as<unsigned>();
+    if (count.max < 1) {
+        throw invalid_argument ("\"max\" must be greater than zero");
+    }
+    if (count.max < count.min) {
+        throw invalid_argument ("\"max\" must be greater than or equal to \"min\"");
+    }
+
+    count.oper = cnode["operator"].as<char>();
+    switch (count.oper) {
+    case '+':
+    case '-':
+    case '^':
+        break;
+    default:
+        throw invalid_argument("Invalid count operator");
+    }
+
```

Comment:
Any reason not to add some validation for the operand  field against the operator? Seems -1 would be an error if the operand is +, 2 would be an error if the operand is -. In general, more validation the less work at the client side...

Oh BTW, what would be the value of max if only min is given?

Reply:
Sure, if someone want to decide what hte operators are going to be, I can add more validation.

Reply:
Oh, and at the moment all four fields are required.  The idea to allowed defaults in the count area is not yet in the RFC, so it doesn't appear in the code yet either.